src/real_ip.rs

```diff
@@ -0,0 +1,299 @@
+use http::{HeaderMap, HeaderValue};
+use ipnetwork::IpNetwork;
+use once_cell::sync::Lazy;
+use std::iter::Iterator;
+use std::net::IpAddr;
+use std::str::from_utf8;
+
+const X_FORWARDED_FOR: &str = "X-Forwarded-For";
+
+const CLOUD_FRONT_STRS: &[&str] = &[
+    // CloudFront IP addresses from http://d7uri8nf7uskq.cloudfront.net/tools/list-cloudfront-ips
+    // Last updated: 2022-03-26
+    "3.10.17.128/25",
```

Comment:
this list was imported from: https://github.com/rust-lang/crates.io/blob/e874a242c5cbff8470bdac62fdf6f05223acc129/config/nginx.conf.erb#L27-L159